src/algebra/covariant_and_contravariant.lean

```diff
@@ -79,13 +79,13 @@ See the `contravariant_class` doc-string for its meaning. -/
 def contravariant : Prop := ∀ (m) {n₁ n₂}, r (μ m n₁) (μ m n₂) → r n₁ n₂
 
 /--  Given an action `μ` of a Type `M` on a Type `N` and a relation `r` on `N`, informally, the
-`covariant_class` says that "the action `μ` preserves the relation `r`.
+`covariant_class` says that "the action `μ` preserves the relation `r`."
 
 More precisely, the `covariant_class` is a class taking two Types `M N`, together with an "action"
 `μ : M → N → N` and a relation `r : N → N → Prop`.  Its unique field `elim` is the assertion that
 for all `m ∈ M` and all elements `n₁, n₂ ∈ N`, if the relation `r` holds for the pair
 `(n₁, n₂)`, then, the relation `r` also holds for the pair `(μ m n₁, μ m n₂)`,
-obtained from `(n₁, n₂)` by "acting upon it by `m`".
+obtained from `(n₁, n₂)` by "acting upon it by `m`.
```

Comment:
Surely this quote is closing the one opened just 20 characters before?
```suggestion
obtained from `(n₁, n₂)` by "acting upon it by `m`".
```

Reply:
(or did you want to remove both, which might be better)

Reply:
Hmm, removing them both also removes the decision about US vs UK punctuation style!
```suggestion
obtained from `(n₁, n₂)` by acting upon it by `m`.
```